accepted/0000-payg-cloud-integration.md

```diff
@@ -0,0 +1,209 @@
+  - authentication header location is an implementation detail
+    - One idea is concatenate in URL as query string parameter and adapt reposync extract and use it
+
+Url to be used should be the one pointing to the cloud RMT server, saved in previous step, instead the one defined in table `susesccrepository`.
+
+### Implementation steps - reposync
+
+Reposync is already able to deal with the basic authentication mechanism (defined in repository URL).
+However needs to be adapted to handle the http authentication header.
+How the header is passed from table `susesccrepositoryauth` to `rhncontentsource` and how is consume by reposync is an implementation detail, as defined in the previous step. We need to maintain this table separation because reposync already have several different mechanism relaying on this separation.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## Product use limitation
+
+Currently SUSE Manager does not prevent misuse of subscriptions, only reports about it (by design).
+
+PAYG instances in Public Clouds are usually limited to one product - using Uyuni to manage them opens the possibility to install a different product, or installing add-ons that are not included the intended price.
+
+It is a legal requirement to have at least minimal UI checks in this regard. Resolving this problem fundamentally (eg. avoiding products aren't changed manually or via Salt) is out of scope.
+
+Note that the Product Migration feature already filters possible migrations correctly. All available migrations can be retrieved via:
+spacewalk-sql --select-mode - <<<"select distinct p1.friendly_name, p2.friendly_name from suseproducts p1 join SUSEUpgradePath up on p1.id = up.from_pdid join suseproducts p2 on p2.id = up.to_pdid order by p1.friendly_name, p2.friendly_name;"
+Current reference list: https://gist.github.com/moio/c008a97620fcc5205b44a0dfd0dc15e9
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Uyuni/SUSE manager PAYG
+
+We could define a uyuni/suse manager PAYG image with access to all cloud RMT repositories.
+It would be possible to syncronize any product directly from cloud in a more simple and straightforward way.
+Drawbacks: User will have access to repository and product he is not paying for.
+
+## Reposync only metadata
+
+Change reposync and SCC to allow metadata access without SCC account to load repositories metadata.
+PAYG client could download data from cloud RMT server directly.
+This will be a huge change, not sure if it is even possible to all providers.
+Possible security impact of having repository metadata publicly accessible.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- How to prevent cross product usage. For example, user with one PAYG SLES-SAP. It can start several PAYG SLES and migrate them to SLES-SAP, paying a smaller fee.
+
+- What happens and how to deal if the PAYG instance we loaded the data from get terminated or stopped?
```

Comment:
In this case the "certificate" become invalid and RMT reject download requests. We should double check this and "test" for it and remove the affected `cloudrmt` auth configs from the Server. Channels will be kept with all the existing content and assigned server will still see it. But no new updates will added to the channels.

Reply:
Since we are using the existing mechanism I think this is already handled by repo sync. More tests need to be done

Reply:
I part of your comment as a note. Thanks